src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -573,7 +573,8 @@ private long transferToDirectlyInternal(long position, int icount,
             do {
                 long comp = Blocker.begin();
                 try {
-                    n = transferTo0(fd, position, icount, targetFD);
+                    boolean append = fdAccess.getAppend(targetFD);
```

Comment:
Can you move this to before the do-while as it does not need to be in the loop or in the Blocker.begin/end.

Reply:
Fixed in e10abeddf7bad5c7ab80c200f78aa3bd5df8916c.